lib/bumblebee/utils/nx.ex

```diff
@@ -325,14 +325,18 @@ defmodule Bumblebee.Utils.Nx do
     Nx.dot(x, [-1], batch_axes, y, [-1], batch_axes)
   end
 
-  defnp normalize(tensor) do
+  defn normalize(tensor) do
     norm =
       tensor
       |> Nx.pow(2)
       |> Nx.sum(axes: [-1], keep_axes: true)
       |> Nx.sqrt()
 
-    tensor / norm
+    if Nx.all(norm) do
+      Nx.divide(tensor, norm)
+    else
+      tensor
+    end
```

Comment:
We normalize each "row" separately, so we need to check individual norms for 0 :)

```suggestion
    norm = Nx.select(norm == 0.0, 1.0, norm)
    
    tensor / norm
```